target/linux/ath79/dts/qca9558_mikrotik_routerboard-922uags-5hpacd.dts

```diff
@@ -47,3 +68,30 @@
 &usb_phy0 {
 	status = "okay";
 };
+
+&mdio1 {
+	status = "okay";
+
+	phy_sfp: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "sgmii";
+		sfp = <&sfp1>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	phy-handle = <&phy_sfp>;
+	pll-data = <0x03000000 0x00000101 0x00001616>;
+	qca955x-sgmii-fixup;
+
+	gmac-config {
+		device = <&gmac>;
+	};
+
+	fixed-link {
```

Comment:
I think it's better to backport upstream phylink support.
ping @olerem

Reply:
Mainline driver is not identical to the downstream driver. In this case it is hard to call it back porting :) But it should be possible. Do any wont to do it?

Reply:
> I think it's better to backport upstream phylink support.
> ping @olerem

Can we do this as an improvement afterwards so this work doesn't get lost?

Reply:
I found https://www.kernel.org/doc/html/latest/networking/sfp-phylink.html, which seems too much for my backporting skills. :disappointed: I think it's reasonable to leave it as a future improvement.